packages/material-ui/src/unstable_useWidth/unstable_useWidth.js

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable camelcase */
+import React from 'react';
+import debounce from 'debounce'; // < 1kb payload overhead when lodash/debounce is > 3kb.
+import { ThemeContext } from '@material-ui/styles/ThemeProvider';
+import { calculateWidth } from '../withWidth/withWidth';
+import createMuiTheme from '@material-ui/core/styles/createMuiTheme';
+
+const defaultTheme = createMuiTheme({
+  typography: {
+    suppressWarning: true,
+  },
+});
+
+export default function unstable_useWidth(options = {}) {
+  const {
+    initialWidth: initialWidthOption,
+    resizeInterval = 166, // Corresponds to 10 frames at 60 Hz.
+    width: widthOption,
+  } = options;
+
+  const theme = React.useContext(ThemeContext) || defaultTheme;
+
+  const [width, setWidth] = React.useState(
+    typeof window !== 'undefined'
+      ? calculateWidth(window.innerWidth, theme.breakpoints)
+      : undefined,
+  );
+
+  React.useEffect(() => {
+    if (typeof window !== 'undefined') {
+      const handleResize = debounce(() => {
+        const newWidth = calculateWidth(window.innerWidth, theme.breakpoints);
+        if (newWidth !== width) {
+          setWidth(newWidth);
+        }
+      }, resizeInterval);
+      window.addEventListener('resize', handleResize);
+      return () => {
+        window.removeEventListener('resize', handleResize);
+      };
+    }
+    return null;
```

Comment:
@oliviertassinari I'm unsure how to test this line since hooks only work if they are fully mounted and if you change `window` to be undefined mount breaks

Reply:
Also I'm unsure how to handle the resize test

Reply:
You can copy & paste the existing withWidth.test.js tests but use the mount API everywhere.

Reply:
withWidth currently relies on the existence of an EventListener component to test resizing. The hook attaches the event to the window

Reply:
Still not sure how to test this

Reply:
I can look into it if you need.

Reply:
Yes, please :)